### DISCUSSION TITLE
Quarkus live debug custom SPI

### DISCUSSION THREAD

**Discussion by MGLL** [keycloak/keycloak#25802] [Q&A] [answered]
Hello,

I have an issue and I'm not able to find the relevant documentation. 
I'm following 

- https://github.com/keycloak/keycloak/blob/main/docs/building.md
- https://github.com/keycloak/keycloak/blob/main/quarkus/README.md

And I have a question, how can I add and live debug a custom SPI into keycloak-quarkus ? 
(with command `../mvnw -f server/pom.xml -Dsuspend=true compile quarkus:dev -Dquarkus.args="start-dev"`)

I tried `java -jar quarkus/server/target/lib/quarkus-run.jar build` but I'm getting an error `"org.keycloak.quarkus.runtime.Environment.getHomePath()" is null`
Or running `kc.sh build` in quarkus.dist.... but I'm getting an error `Error: Could not find or load main class io.quarkus.bootstrap.runner.QuarkusEntryPoint`

So is there a way to add and debug a custom SPI this way ?
Or should I put my SPI in the `quay.io/keycloak/keycloak` container, run it on dev mode and plug my remote debugger on it ? 

(I'm using Intellij IDEA Ultimate, if there is a configuration to do)
Thanks in advance

**Comment by MGLL** [comment, marked as answer]
I have found the solution, which is basically relying on remote debug with the custom SPI bundled into the keycloak container.

**Reply by codespearhead** [reply]
@MGLL an you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so it doesn't show up in the discussion tab as unanswered?
